app/course/services/courseStateService.js

```diff
@@ -490,10 +490,19 @@ courseApp.service('courseStateService', function ($rootScope, $log, Course, Term
 					uiState.massImportPrivate = false;
 					return uiState;
 				case REMOVE_COURSE:
+					var courseId = action.payload.course.id;
 					// Remove the details pane if it was showing the deleted course
-					if (uiState.selectedCourseId == action.payload.course.id) {
+					if (uiState.selectedCourseId == courseId) {
 						uiState.selectedCourseId = null;
 					}
+
+					// Ensure deleted course is not selected
+					var index = uiState.selectedCourseRowIds.indexOf(courseId);
```

Comment:
I'd just combine these two lines (500 and 502) to get rid of the temporary variable. Readability wouldn't be that bad.